### PR TITLE
common.py: Fix level function

### PR DESCRIPTION
**Pull request by RHanley1**

### CODE REVIEW

**Comment by RHanley1** [comment]
Can we merge this also otherwise the ```ac_zeeman_shift``` function #33 won't work. AFAICT it is only the ```ac_zeeman_shift``` function which uses the ```ion.level()``` function

**Comment by hartytp** [comment]
> Can we merge this also otherwise the ac_zeeman_shift function #33 won't work.

No. It's not good to get into the habit of merging commits in a rush to solve a short-term need like this. I want to find a proper solution. Even if it's only used in one place right now, that will likely change (and if it doesn't then the function is probably overly specialized and should be removed).

I'll have a think about how I want to solve this shortly. In the meantime, if you want to get your code working quickly then just run off a branch.

**Comment by RHanley1** [comment]
Fair enough, I was running off my own branch anyway; was thinking more for others. Out of interest, why is this not a proper solution? I couldn't find a way of comparing a number with a slice object without explicitly breaking it apart.

**Comment by hartytp** [comment]
> Out of interest, why is this not a proper solution? I couldn't find a way of comparing a number with a slice object without explicitly breaking it apart.

The current design essentially assumes that the start/stop indices are private to a LevelData. This method works around that in a really ugly way, suggesting that the interface is not great. There are a few potentially better solutions: move this functionality into `LevelData`, make the start/stop indices public interface, etc

As I said above, I also suspect that testing if the state is in the range is needlessly slow compared with a simple bounds check (state >= start and state < stop). in this instance the speed is probably not particularly relevant but if one gets in the habit of making everything needlessly slow then the optimization etc can become really painful. I've tried to design everything to be decently efficient.

File: ion_phys/common.py [hunk 1]
```diff
@@ -183,7 +183,8 @@ def index(self, level, M, *, F=None, MI=None, MJ=None):
     def level(self, state):
         """ Returns the level a state lies in. """
         for level, data in self.levels.items():
-            if state in data.slice():
+            sl = data.slice()
+            if state in range(sl.start, sl.stop):
```
Comment (pathfinder49) [inline ion_phys/common.py:187]: I think we should get a version of this merged. Would changing this to 
```python3
if state > min(sl.start, sl.stop) and state < max(sl.start, sl.stop):
```
seems like a reasonable compromise.

**Comment by hartytp** [comment]
@pathfinder49 is this causing issues in the lab right now? The real solution here is to implement the small api improvements suggested elsewhere, which is something that's been on my to do list for a while. I don't mind merging your solution if you need this stat (and can't just work off a branch for now) but I want to keep a tracking issue open to remind myself to implement a better solution in due course...

**Comment by pathfinder49** [comment]
When calculating AC-Zeeman shifts we are currently forced to use Ryans branch which makes it unsuitable for calculations in experiments. Ryans branch is also falling behind ion_phys/master, requiring more work for others to get all features.
The API improvements seem like a good idea. However, I don't believe they should hold up this PR.

**Comment by hartytp** [comment]
Fair enough. Feel free to merge your solution, but can you create a new issue that references this PR and the commit you merge in so I know what to look at when I get round to it?

**Pull request closed without merging**